azure-iot-device/setup.cfg

```diff
@@ -5,8 +5,36 @@
 #--------------------------------------------------------------------------
 
 [bdist_wheel]
-# This flag says to generate wheels that support both Python 2 and Python
-# 3. If your code will not run unchanged on both Python 2 and 3, you will
-# need to generate separate wheels for each Python version that you
+# This flag says to generate wheels that support both Python 2 and Python 
+# 3. If your code will not run unchanged on both Python 2 and 3, you will 
+# need to generate separate wheels for each Python version that you 
 # support.
-universal=1
+universal = 1
+
+[bumpversion]
+current_version = 2.0.0-preview.7
+parse = (?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)-preview\.(?P<num>\d+)
+serialize = 
+	{major}.{minor}.{patch}-preview.{num}
+
+[bumpversion:part:num]
+first_value = 0
+values = 
+	0
+	1
+	2
+	3
+	4
+	5
+	6
+	7
+	8
+	9
+	10
+	11
+	12
+	13
+	14
+	15
+	16
```

Comment:
>16 [](start = 1, length = 2)

are these values for major minor patch and num ? are  we okay to bump till 16 for the patch field specially ?

Reply:
I'll update the documentation on this one and repost

Reply:
The values are for num.  I updated the internal documentation about this section so please check that out (https://github.com/Azure/iot-sdks-internals/pull/105)

Based on this (https://www.python.org/dev/peps/pep-0386/) which says, "In Python there are no real restrictions yet on how a project should manage its versions, and how they should be incremented.", we are ok to bump until version 16 (and beyond if you choose).  But I'm no Python dev - if you have other information please share.

Reply:
I have no specific information except that I have seen the python versions itself be bumped to 10 or 16. For example python 2.7 started from 2.7.0  and ended at 2.7.16 , 
python 3.4 started at 3.4.0 and ended at 3.4.10. 
These are the examples where the incremental part though. So 16 seems fine to be a choice.

If it is for num (so the part after preview) I don't think it matters that much as it will no longer be there when we stop having preview.



---
In reply to: [302317534](https://github.com/Azure/azure-iot-sdk-python-preview/pull/134#discussion_r302317534) [](ancestors = 302317534)

Reply:
I am also confused by this list of numbers. This num value can go arbitrarily high, we could theoretically get up to 2.0.0-preview.35342

Why don't we just increment num rather than picking it from a list of defined values?

---
In reply to: [302766600](https://github.com/Azure/azure-iot-sdk-python-preview/pull/134#discussion_r302766600) [](ancestors = 302766600,302317534)

Reply:
I agree this approach seems awakward but it is by design.

The author give examples of enumerated strings for this configruation: alpha, beta, gamma, which are arbitrarily sorted as the user prefers.  In their documentation the author's stated intent is maximum flexibility.  Semver complainace is not mentioned and I suspect the tool predates semver but I'm not sure.

As Oliva mentions, this section will be removed when the -preview label is deprecated and the major, minor, patch values will increment as expected.

Source code: https://github.com/peritus/bumpversion

Reply:
I took a look at the docs, and it seems the default values of the `values` attribute is already numeric incrementation, and that setting specified values is for using strings to indicate a named version. Can you double check that this is even necessary? My understanding from reading them is that you could remove these defined values and it should still work.

---
In reply to: [303091690](https://github.com/Azure/azure-iot-sdk-python-preview/pull/134#discussion_r303091690) [](ancestors = 303091690)

Reply:
You are right - I tested and it works; apologies for my earlier confusion.  But in this testing I discovered a new problem - the tool strips the comments from setup.cfg as you noted so I'll need to make a separate file, test this one, and update the pr.  This file will be reverted.